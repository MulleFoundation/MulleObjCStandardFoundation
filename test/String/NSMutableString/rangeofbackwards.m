//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"


static void  print_range( NSRange range)
{
   if( range.location == NSNotFound && range.length == 0)
      printf( "{ NSNotFound, 0 }\n");
   else
      printf( "{ %lu, %lu }\n", (long) range.location, (long) range.length);
}


int main(int argc, const char * argv[])
{
   NSMutableString     *s;
   NSRange             range;

   s = [NSMutableString object];

   [s appendString:@"VfL"];
   [s appendString:@" "];
   [s appendString:@"Bochum"];
   [s appendString:@" "];
   [s appendString:@"VfL 1848"];

   range = [s rangeOfString:@"L Bo"
                    options:NSBackwardsSearch];
   print_range( range);

   range = [s rangeOfString:@"1848"
                    options:NSBackwardsSearch];
   print_range( range);

   range = [s rangeOfString:@"VfL"
                    options:NSBackwardsSearch];
   print_range( range);

   range = [s rangeOfString:@"BOCHUM"
                    options:NSBackwardsSearch];
   print_range( range);

   range = [s rangeOfString:@"BOCHUM"
                    options:NSBackwardsSearch|NSCaseInsensitiveSearch];
   print_range( range);

   return( 0);
}
