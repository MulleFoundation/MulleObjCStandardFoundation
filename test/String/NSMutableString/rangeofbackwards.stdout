{ 2, 4 }
{ 15, 4 }
{ 11, 3 }
{ NSNotFound, 0 }
{ 4, 6 }
